client/widgets/markets/CTradeBase.cpp

```diff
@@ -0,0 +1,111 @@
+/*
+ * CTradeBase.cpp, part of VCMI engine
+ *
+ * Authors: listed in file AUTHORS in main folder
+ *
+ * License: GNU General Public License v2.0 or later
+ * Full text of license available in license.txt file, in main folder
+ *
+ */
+#include "StdInc.h"
+#include "CTradeBase.h"
+
+#include "../MiscWidgets.h"
+
+#include "../../gui/CGuiHandler.h"
+#include "../../widgets/Buttons.h"
+#include "../../widgets/TextControls.h"
+
+#include "../../CGameInfo.h"
+
+#include "../../../lib/CGeneralTextHandler.h"
+#include "../../../lib/mapObjects/CGHeroInstance.h"
+
+CTradeBase::CTradeBase(const IMarket * market, const CGHeroInstance * hero)
+	: market(market)
+	, hero(hero)
+{
+	deal = std::make_shared<CButton>(Point(), AnimationPath::builtin("ALTSACR.DEF"),
+		CGI->generaltexth->zelp[585], std::bind(&CTradeBase::makeDeal, this));
```

Comment:
This button should actually be different for creature / artifacts altars:
`CGI->generaltexth->zelp[584]` -> creatures altar
`CGI->generaltexth->zelp[585]` -> artifacts altar

Identical for English, but different in some languages. Was reported on Discord recently